client-common/src/tendermint/types/block.rs

```diff
@@ -33,18 +33,38 @@ pub struct Header {
 }
 
 impl Block {
-    /// Returns transactions in a block (this may also contain invalid transactions)
-    pub fn transactions(&self) -> Result<Vec<TxAux>> {
+    /// Returns un-encrypted transactions in a block (this may also contain invalid transactions)
+    ///
+    /// NOTE: Un-encrypted transactions only contain deposit stake and unbond stake transactions
+    pub fn unencrypted_transactions(&self) -> Result<Vec<Transaction>> {
         match &self.block.data.txs {
             None => Ok(Vec::new()),
-            Some(txs) => txs
-                .iter()
-                .map(|raw_tx| Ok(decode(&raw_tx).context(ErrorKind::DeserializationError)?))
-                .map(|bytes: Result<Vec<u8>>| {
-                    Ok(TxAux::decode(&mut bytes?.as_slice())
-                        .ok_or(ErrorKind::DeserializationError)?)
-                })
-                .collect::<Result<Vec<TxAux>>>(),
+            Some(transactions) => {
+                let unencrypted_transactions = transactions
+                    .iter()
+                    .map(|raw_transaction| -> Result<Option<Transaction>> {
+                        let decoded =
+                            decode(&raw_transaction).context(ErrorKind::DeserializationError)?;
+                        let tx_aux = TxAux::decode(&mut decoded.as_slice())
+                            .ok_or_else(|| Error::from(ErrorKind::DeserializationError))?;
+
+                        match tx_aux {
+                            TxAux::DepositStakeTx { tx, .. } => {
+                                Ok(Some(Transaction::DepositStakeTransaction(tx)))
+                            }
+                            TxAux::UnbondStakeTx(tx, _) => {
+                                Ok(Some(Transaction::UnbondStakeTransaction(tx)))
+                            }
+                            _ => Ok(None),
+                        }
+                    })
+                    .collect::<Result<Vec<Option<Transaction>>>>()?;
```

Comment:
could the `filter_map` directly on the first iter()?

Reply:
Done